## API : 
        Your code -->Api call-->Server-->Json-->Your COde
    1. URL : restaurant Menu .
    2. GET : Menu
    3. Post : Place an order
    4. Status Code : Order sucess/order failed

### Basic Structure of api :
                            1. Endpoint url : The adress of the service .
                            2. Http Method : Get,Post(Put ,Patch,Delete)
                            3. Headers->Metadata such as authentication .
                            4. Paarameter or body:
                                                    a. query parameter
                                                    b. Path Parameter
                                                    c. Body

In [3]:
import requests

response=requests.get("https://api.github.com/")

print(response.text)

print(response.status_code)

{"current_user_url":"https://api.github.com/user","current_user_authorizations_html_url":"https://github.com/settings/connections/applications{/client_id}","authorizations_url":"https://api.github.com/authorizations","code_search_url":"https://api.github.com/search/code?q={query}{&page,per_page,sort,order}","commit_search_url":"https://api.github.com/search/commits?q={query}{&page,per_page,sort,order}","emails_url":"https://api.github.com/user/emails","emojis_url":"https://api.github.com/emojis","events_url":"https://api.github.com/events","feeds_url":"https://api.github.com/feeds","followers_url":"https://api.github.com/user/followers","following_url":"https://api.github.com/user/following{/target}","gists_url":"https://api.github.com/gists{/gist_id}","hub_url":"https://api.github.com/hub","issue_search_url":"https://api.github.com/search/issues?q={query}{&page,per_page,sort,order}","issues_url":"https://api.github.com/issues","keys_url":"https://api.github.com/user/keys","label_searc

response.text : gives the simple text form of the api response
reponse.status_code : Tells about weather the fetching being successfull or not .
requests.get: get the data from url .
(200 is the success message .)

Tring to give the wring url 


In [4]:
response=requests.get("https://api.github.com/invalid")
print(response.text)
print(response.status_code)

{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest",
  "status": "404"
}
404


stattus_code : 404 ,
and printed the not found response

In [5]:
response=requests.get("https://api.github.com/")

data=response.json()
print(data["current_user_url"])


https://api.github.com/user
